platform/android/export/export.cpp

```diff
@@ -2101,13 +2102,23 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			}
 		}
 
+		if (bool(p_preset->get("custom_template/export_app_bundle"))) {
+			if (!bool(p_preset->get("custom_template/use_custom_build"))) {
+				valid = false;
+				err += TTR("\"Export App Bundle\" is only valid when \"Use Custom Build\" is enabled.");
+				err += "\n";
+			}
+		}
+
 		r_error = err;
 		return valid;
 	}
 
 	virtual List<String> get_binary_extensions(const Ref<EditorExportPreset> &p_preset) const {
 		List<String> list;
 		list.push_back("apk");
+		list.push_back("aab");
+		//TODO: make sure "aab" is selected only when "export_app_bundle" is selected.
```

Comment:
Can you just add the check here? Or is this function called and its result cached across toggling of the AAB checkbox?

Reply:
All this does is adds ".aab" as a valid choice for the filename extension. The check has to be made in the `export_project` function, but I'm still figuring out where to put it.

Reply:
I think I took care of this now [here](https://github.com/amanj120/godot/pull/18/files#diff-fb88394f2df1faf844deaff3b3639aeeR2468)